### ISSUE TITLE
Documentation is not up to date for master branch

### ISSUE THREAD

**Issue by beckyconning** [purescript-halogen/purescript-halogen#588]
https://github.com/slamdata/purescript-halogen/blob/master/docs/5%20-%20Parent%20and%20child%20components.md

Is misleading as the references to `ParentHTML` and `ParentDSL` give the impression that there is a difference between parent and child components.

**Comment by garyb** [comment]
Yeah, the documentation is all pretty wrong for the code in master - it will essentially need a rewrite before the next release is made. The plan is to pre-release after #576, and then make a proper release once the docs are updated too.

**Comment by JordanMartinez** [comment]
(I'm posting this in other doc-related issues. Sorry for the multiple notifications.)

Just FYI. I've put together a bottom-up tutorial on Halogen for the `v5.0.0-rc.4` release: https://github.com/jordanmartinez/learn-halogen There are still a few things it doesn't cover yet, but it should get most people up to speed.

**Comment by Benjmhart** [comment]
I'm getting set up to do a bunch of documentation updates in the coming weeks, hopefully will be able to bring the docs fully up to date for v5

**Comment by garyb** [comment]
I do have a whole new set of documentation in progress, I had been making good progress with it, but have a deadline this week so it's been on hold again. If you get there before I finish, I'll definitely take it though! It would make some things easier, as there are a few changes stacking up that I might want to make as a v6 rather than having a confusing distinction between v5s, so I could drop my new document then instead.

**Comment by Benjmhart** [comment]
I'm about 3 chapters in,  so hoping to have the existing chapters replaced with v5 docs within another two weeks depending on my availability.

**Comment by Benjmhart** [comment]
if we get this merged in,  is there anything else holding back v5 from official release?   part of my motivation is getting the Pursuit docs updated to v5 eventually. @garyb

**Comment by garyb** [comment]
Yeah, the docs are the only thing holding back a v5 release.

**Comment by Benjmhart** [comment]
cool : )

**Comment by sigilion** [comment]
Have you made much progress? I might have time to jump in if there's still work to do.

**Comment by Benjmhart** [comment]
Hey,  sorry for the delay I've had some travel for work,   I'm revising the
last chapter to bring us to documentation parity with v4 right now, I'm
hoping to PR that by Sunday or Monday.

I have some additions, but I think those are best left to a follow-up PR.

how does that sound?

Benjamin James Mikel Hart

**Comment by sigilion** [comment]
Sounds good to me, I just had some free time this week and thought I'd offer to help :). I had a go at chapter 5 anyway, my (very rough) effort [here](https://github.com/sigilion/purescript-halogen/commit/4c5e338d61583df8d8c28f0fd27802c0d344445d) if that is of any help, but feel free to ignore.
